src/reflect/type.go

```diff
@@ -631,6 +632,93 @@ func rawStructFieldFromPointer(descriptor *structType, fieldType *rawType, data
 	}
 }
 
+const fieldOffsetCacheSize = 16
```

Comment:
We could probably scale this like
```
const fieldOffsetCacheSze = 4 * unsafe.Sizeof(uintptr(0))
```
so when building natively or on larger machines we get a larger default cache.  I was just trying to be really conservative here.